### DISCUSSION TITLE
COPY NOT ALLOWING ON BOUNDARY_CONDITIONS?

### DISCUSSION THREAD

**Discussion by govindiitd** [ansys/pyfluent#2297] [General]
Hi,

I am starting to experience pyfluent with "cavitation" example.  I get the following error message for 
solver.setup.boundary_conditions.copy(from_="inlet_1", to="inlet_2"), 

AttributeError: "copy" is not an allowed Settings objects name.

Regards,
Govind

**Comment by RobPasMue** [comment]
Transferring this discussion to the PyFluent repository

**Comment by ansjsia** [comment]
@govindiitd Could you share the following information?

- Which version(s) of ANSYS Fluent do you have installed?
- Which version of PyFluent do you have installed and how did you install it? (i.e. using `pip install` or by cloning from GitHub?)
- How are you running the example? Are you running it as a script or as a Jupyter Notebook?

**Comment by govindiitd** [comment]
Hi,

-> Ansys version Fluent: 2023R1
-> pyFluent Version: ansys-fluent-core 0.18.2, and installed using pip
-> I am running it as a jupyter-notebook


Regards,
Govind

**Comment by ansjsia** [comment]
Hi Govind,

Calling `solver.setup.boundary_conditions.command_names` or `solver.setup.boundary_conditions.get_active_command_names()` shows that the `solver.setup.boundary_conditions.copy()` PyFluent settings API method is not available in 23R1. If you are not able to upgrade to 23R2, you could replace the settings API calls with the equivalent TUI command.

For example, to copy a boundary condition, you can run the TUI command `define/boundary-conditions/copy-bc inlet_1 inlet_2 ()`. This can be called through PyFluent by  `solver.tui.define.boundary_conditions.copy_bc("inlet_1", "inlet_2", ())`.

Thanks,
Jess

**Comment by govindiitd** [comment]
Hi,

Thanks, letting me know about this. It works fine for me and I am able to
manage some other stuff as well.

Other thing:

"solver.results.graphics.contour["contour_vf_vapor"].display()" not
displaying any image in notebook. What should I do?

Regards,
Govind

**Reply by ansjsia** [reply]
Calling the `.display()` method on a `contour` object will display it in the Fluent GUI (it would be visible if you launched Fluent with the argument `show_gui=True`), not in the notebook.

To post-process in the notebook, I would suggest looking into PyFluent Visualization:
https://visualization.fluent.docs.pyansys.com/version/dev/examples/index.html
https://examples.fluent.docs.pyansys.com/version/dev/examples/00-released_examples/01-brake.html
